editor/src/components/inspector/inspector-strategies/inspector-strategy.tsx

```diff
@@ -0,0 +1,25 @@
+import { ElementInstanceMetadataMap } from '../../../core/shared/element-template'
+import { ElementPath } from '../../../core/shared/project-file-types'
+import { CanvasCommand } from '../../canvas/commands/commands'
+import { EditorDispatch } from '../../editor/action-types'
+import { applyCommandsAction } from '../../editor/actions/action-creators'
+
+export type InspectorStrategy = (
+  metadata: ElementInstanceMetadataMap,
+  selectedElementPaths: Array<ElementPath>,
+) => Array<CanvasCommand> | null
+
+export function runStrategies(
+  dispatch: EditorDispatch,
+  metadata: ElementInstanceMetadataMap,
+  selectedViews: ElementPath[],
+  strategies: InspectorStrategy[],
+): void {
+  for (const strategy of strategies) {
+    const commands = strategy(metadata, selectedViews)
+    if (commands != null) {
+      dispatch([applyCommandsAction(commands)])
+    }
+    return
```

Comment:
Doesn't this `return` statement cause it to exit after the first `strategy` call?

Reply:
this is fully intentional, the strategies should be in order of specificity, with the most specific matching one applied. As far as I understand this is the same principle that the canvas strategies use (with the difference that inspector strategies don't have a strategy picker (yet))